### PR TITLE
lightning: add parameter to output logs of all packages and enable grpc log

### LINKED ISSUE #45497: lightning, enable logs about client-go/tidb and grpc to help diagnose
## Enhancement
lightning, enable logs about client-go/tidb and grpc to help diagnose some problems

will add a parameter `enable-diagnose-logs`, default false, to keep old log behavior and avoid output too much logs on normal run

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #45497

Problem Summary:

### What is changed and how it works?
- add parameter `enable-diagnose-logs` to output logs of all packages and enable grpc log when enabled, default is false
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - enable the parameter, we can see logs: `coprocessor.go` of tidb, and `client_batch.go` of client-go
```log
[2023/07/21 14:48:09.126 +08:00] [INFO] [grpclogger.go:92] ["[core][Channel #1] Channel created"] [system=grpc] [grpc_log=true]
[2023/07/21 14:48:09.126 +08:00] [INFO] [grpclogger.go:92] ["[core][Channel #1] original dial target is: \"127.0.0.1:2379\""] [system=grpc] [grpc_log=true]


[2023/07/21 14:16:56.613 +08:00] [INFO] [coprocessor.go:1290] ["[TIME_COP_PROCESS] resp_time:380.626041ms txnStartTS:443002375548698627 region_id:128 store_addr:192.168.192.77:20160 kv_process_ms:194 kv_w
ait_ms:5 kv_read_ms:0 processed_versions:5072 total_versions:5073 rocksdb_delete_skipped_count:0 rocksdb_key_skipped_count:10143 rocksdb_cache_hit_count:2 rocksdb_read_count:2542 rocksdb_read_byte:2460442
7"]

[2023/07/21 14:20:27.622 +08:00] [INFO] [import.go:1336] [progress] [total=100.0%] [tables="2/2 (100.0%)"] [chunks="2/2 (100.0%)"] [engines="4/4 (100.0%)"] [restore-bytes=489.6MiB/489.6MiB] [restore-rows=
25004/25004(estimated)] [import-bytes=552.1MiB/552.1MiB(estimated)] ["encode speed(MiB/s)"=0.6799745656866469] [state=post-processing] []
[2023/07/21 14:20:27.623 +08:00] [DEBUG] [client_batch.go:615] ["batchRecvLoop fails when receiving, needs to reconnect"] [target=192.168.192.77:20160] [forwardedHost=] [error="rpc error: code = Unavailab
le desc = keepalive watchdog timeout"]
[2023/07/21 14:20:43.450 +08:00] [DEBUG] [client_batch.go:615] ["batchRecvLoop fails when receiving, needs to reconnect"] [target=192.168.192.77:20160] [forwardedHost=] [error="rpc error: code = Unavailab
le desc = keepalive watchdog timeout"]
[2023/07/21 14:20:43.453 +08:00] [DEBUG] [pd.go:100] ["store min resolved ts"] [resp="{\n  \"is_real_time\": true,\n  \"min_resolved_ts\": 443002498349793281,\n  \"persist_interval\": \"1s\"\n}\n"]
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by D3Hunter** [comment]
ci pass https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_br_integration_test/detail/pull_br_integration_test/296/pipeline

File: br/pkg/lightning/log/log.go [hunk 1]
```diff
@@ -77,10 +80,24 @@ var (
 
 // InitLogger initializes Lightning's and also the TiDB library's loggers.
 func InitLogger(cfg *Config, _ string) error {
+	loggerOptions := []zap.Option{}
+	if cfg.EnableDiagnoseLogs {
+		// the value doesn't matter, logutil.InitLogger only checks whether it's empty.
+		if err := os.Setenv(logutil.GRPCDebugEnvName, "true"); err != nil {
+			L().Warn("failed to enable GRPC debug log", zap.Error(err))
```
Comment (lance6716) [inline br/pkg/lightning/log/log.go:87]: L() is not initialized here

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**